disruptor/src/main/java/org/axonframework/disruptor/commandhandling/CommandHandlerInvoker.java

```diff
@@ -293,10 +293,10 @@ private DisruptorRepository(AggregateFactory<T> aggregateFactory, Cache cache, E
                                                                                   repositoryProvider,
                                                                                   trigger);
 
-            CurrentUnitOfWork.get().onCommit(u-> {
+            if (aggregate.identifierAsString() != null) {
```

Comment:
Although the check isn't magical, why we need to do it is rather specific. Would you mind adding a comment just above the if-block explaining why this validation is done?
So, something along the lines specifying that on a create-or-update the aggregate identifier isn't set yet for a Command Model serviced by the Disruptor Command Bus. Hence, we only set it in the cache if it's not-null.